compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -1312,12 +1296,11 @@ impl Type {
         let this = self.follow_bindings();
         let target = target.follow_bindings();
 
-        if let (Type::Array(size1, element1), Type::Array(size2, element2)) = (&this, &target) {
-            let size1 = size1.follow_bindings();
-            let size2 = size2.follow_bindings();
+        if let (Type::Array(size, element1), Type::Slice(element2)) = (&this, &target) {
+            let length = size.follow_bindings();
 
             // If we have an array and our target is a slice
-            if matches!(size1, Type::Constant(_)) && matches!(size2, Type::NotConstant) {
+            if matches!(length, Type::Constant(_))  {
```

Comment:
I think this check can be removed entirely since we already know we have an array and a slice